packages/protocol/contracts/common/FeeHandler.sol

```diff
@@ -181,48 +237,80 @@ contract FeeHandler is
     tokenState.active = true;
     tokenState.handler = handlerAddress;
 
-    activeTokens.push(tokenAddress);
+    activeTokens.add(tokenAddress);
   }
 
-  function addToken(address tokenAddress, address handlerAddress) external onlyOwner {
-    _addToken(tokenAddress, handlerAddress);
+  /**
+    @dev Deactivates the specified token by marking it as inactive.
+    @param tokenAddress The address of the token to deactivate.
+  */
+  function deactivateToken(address tokenAddress) external onlyOwner {
+    _deactivateToken(tokenAddress);
+  }
+
+  function _deactivateToken(address tokenAddress) private {
+    activeTokens.remove(tokenAddress);
+    TokenState storage tokenState = tokenStates[tokenAddress];
+    tokenState.active = false;
+  }
+
+  // TODO test me
+  function setHandler(address tokenAddress, address handlerAddress) external onlyOwner {
+    _setHandler(tokenAddress, handlerAddress);
+  }
+
+  // TODO test me
+  function _setHandler(address tokenAddress, address handlerAddress) private {
+    require(handlerAddress != address(0), "Can't set handler to zero, use deactivateToken");
+    TokenState storage tokenState = tokenStates[tokenAddress];
+    tokenState.handler = handlerAddress;
   }
 
   function getActiveTokens() public view returns (address[] memory) {
-    return activeTokens;
+    return activeTokens.values;
   }
 
-  function removeToken(address tokenAddress) external {}
+  function removeToken(address tokenAddress) external onlyOwner {
+    _removeToken(tokenAddress);
+  }
+
+  function _removeToken(address tokenAddress) private {
+    _deactivateToken(tokenAddress);
+    TokenState storage tokenState = tokenStates[tokenAddress];
+    tokenState.handler = address(0);
```

Comment:
This can only be executed by governance and the amount of tokens is quite limited, so I'd prefer to have this easier to query rather than get the storage back.